### PR TITLE
Assure Responsive Image Generation Respects Conversion Quality Setting

### PR DESCRIPTION
**Pull request by MattStrauss**
I saw the help wanted tag on [this issue](https://github.com/spatie/laravel-medialibrary/issues/1862), and I wanted to contribute as all of the awesome Spatie packages are such a boon to the Laravel community. 

## Original Issue
The [original issue](https://github.com/spatie/laravel-medialibrary/issues/1862) was that the the responsive image generation process did not respect the quality setting of a conversion. 

## Proposed Solution
This pull request adds a private method to determine if the conversion has a quality setting, and if so, it applies that quality setting to the generated responsive images. 

## Tests
I also created a test for the change to assure it actually worked. I wanted to be more precise, i.e. show that at 60% quality the image would be 60% smaller, but while the numbers are close, they're not exact. So I simply asserted that the lower quality responsive image should have a smaller file size than the standard responsive image. 

## Concerns 
I added two additional conversions in `TestModelWithResponsiveImages`, and I'm hoping this doesn't slow the test suite down too much. I looked for alternative ways to test the changes but all of the current conversions within this class didn't quite unambiguously achieve what the test needed to prove, IMO. 

If any changes are needed or this just isn't the solution you are looking for, don't hesitate to let me know, I will take no offense. And thanks again for all of your amazing packages!

### CODE REVIEW

File: src/ResponsiveImages/ResponsiveImageGenerator.php [hunk 1]
```diff
@@ -64,18 +64,30 @@ public function generateResponsiveImagesForConversion(Media $media, Conversion $
...
+        }
+
+        return $conversion->getManipulations()->getManipulationArgument('quality') ?: $defaultConversionQuality;
+    }
+
     public function generateResponsiveImage(
         Media $media,
         string $baseImage,
         string $conversionName,
+        int $conversionQuality,
```
Comment (freekmurze) [inline src/ResponsiveImages/ResponsiveImageGenerator.php:90]: Addin a parameter here is a breaking change.

Comment (MattStrauss) [reply src/ResponsiveImages/ResponsiveImageGenerator.php:90]: Sure thing, I'll fix that shortly, likely just make it an optional parameter.

Comment (MattStrauss) [reply src/ResponsiveImages/ResponsiveImageGenerator.php]: See the [newest commit](https://github.com/spatie/laravel-medialibrary/pull/1881/commits/bd3b36a27380628d38ed21baa4e1f37ddb41ba35) that (hopefully) achieved the changes you requested.

**Comment by freekmurze** [comment]
This looks better 👍 

Could you also update the docs with an example of the new functionality?

**Comment by MattStrauss** [comment]
I updated the docs per @freekmurze request. Let me know if that makes sense, or if you prefer that I add a small **standalone** section in the docs for this. I just tacked on some details to the most pertinent existing section as this seems like a smallish change/feature.

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**